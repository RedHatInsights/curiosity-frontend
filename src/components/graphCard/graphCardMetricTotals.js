import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardBody, CardFooter, CardHeader, CardTitle, Title } from '@patternfly/react-core';
import _camelCase from 'lodash/camelCase';
import { useProductGraphTallyQuery } from '../productView/productViewContext';
import { useGraphCardContext, useMetricsSelector } from './graphCardContext';
import { Loader, SkeletonSize } from '../loader/loader';
import { toolbarFieldOptions } from '../toolbar/toolbarFieldRangedMonthly';
import { RHSM_API_QUERY_SET_TYPES } from '../../services/rhsm/rhsmConstants';
import { helpers } from '../../common';

/**
 * @memberof GraphCard
 * @module GraphCardMetricTotals
 */

/**
 * Display totals for a single metric.
 *
 * @param {object} props
 * @param {React.ReactNode} props.children
 * @param {Function} props.useGraphCardContext
 * @param {Function} props.useMetricsSelector
 * @param {Function} props.useProductGraphTallyQuery
 * @returns {React.ReactNode}
 */
const GraphCardMetricTotals = ({
  children,
  useGraphCardContext: useAliasGraphCardContext,
  useMetricsSelector: useAliasMetricsSelector,
  useProductGraphTallyQuery: useAliasProductGraphTallyQuery
}) => {
  const { settings = {} } = useAliasGraphCardContext();
  const { [RHSM_API_QUERY_SET_TYPES.START_DATE]: startDate } = useAliasProductGraphTallyQuery();
  const { pending, error, fulfilled, dataSets = [] } = useAliasMetricsSelector();
  const { data = [], id: firstChartId, metric: firstMetricId, meta = {} } = dataSets[0] || {};
  const { date: lastDate, hasData: lastHasData, y: lastValue } = data[data.length - 1] || {};

  const {
    date: currentDate,
    hasData: currentHasData,
    y: currentValue
  } = data.find(({ isCurrentDate }) => isCurrentDate === true) || {};

  const { totalMonthlyDate: monthlyDate, totalMonthlyHasData: monthlyHasData, totalMonthlyValue: monthlyValue } = meta;

  const { title: selectedMonth, isCurrent } =
    toolbarFieldOptions.find(
      option => option.title === startDate || option.value.startDate.toISOString() === startDate
    ) || {};

  const dailyDate = isCurrent ? currentDate : lastDate;
  const dailyHasData = isCurrent ? currentHasData : lastHasData;
  const dailyValue = isCurrent ? currentValue : lastValue;

  if (settings?.isMetricDisplay && settings?.cards?.length) {
    const metricDisplayPassedData = helpers.setImmutableData({
      chartId: firstChartId,
      dailyDate,
      dailyHasData,
      dailyValue,
      metricId: firstMetricId,
      groupMetricId: [...settings.groupMetric],
      monthlyDate,
      monthlyHasData,
      monthlyValue,
      selectedValue: selectedMonth
    });

    return (
      <div
        data-test={`graphMetricTotals-${settings?.groupMetric?.map(metricId => _camelCase(metricId))?.join('-')}`}
        data-test-data={JSON.stringify(metricDisplayPassedData)}
        className="curiosity-usage-graph__totals"
      >
        <div>
          <div className="curiosity-usage-graph__totals-column">
            {settings?.cards?.map(({ key, header, body, footer }, index) => (
              <Card
                key={key || helpers.generateHash({ metricDisplayPassedData, index })}
                isPlain
                data-test={`graphMetricTotalsCard-${index}`}
                className={`curiosity-usage-graph__totals-column-card ${(error && 'blur') || ''}`}
              >
                <CardHeader>
                  <CardTitle>
                    <Title headingLevel="h2" size="md">
                      {pending && <Loader variant="skeleton" skeletonProps={{ size: SkeletonSize.lg }} />}
                      {fulfilled && ((typeof header === 'function' && header(metricDisplayPassedData)) || header)}
                    </Title>
                  </CardTitle>
                </CardHeader>
                <CardBody>
                  <div>
                    {pending && <Loader variant="skeleton" skeletonProps={{ size: SkeletonSize.lg, height: '60px' }} />}
                    {fulfilled && ((typeof body === 'function' && body(metricDisplayPassedData)) || body)}
                  </div>
                </CardBody>
                <CardFooter>
                  <div>
                    {pending && <Loader variant="skeleton" skeletonProps={{ size: SkeletonSize.lg }} />}
                    {fulfilled && ((typeof footer === 'function' && footer(metricDisplayPassedData)) || footer)}
                  </div>
                </CardFooter>
              </Card>
            ))}
          </div>
        </div>
        <div>
          <div className="curiosity-usage-graph__totals-graph-column">{children}</div>
        </div>
      </div>
    );
  }

  return children;
};

/**
 * Prop types.
 *
 * @type {{useProductGraphTallyQuery: Function, children: React.ReactNode, useMetricsSelector: Function}}
 */
GraphCardMetricTotals.propTypes = {
  children: PropTypes.node,
  useGraphCardContext: PropTypes.func,
  useMetricsSelector: PropTypes.func,
  useProductGraphTallyQuery: PropTypes.func
};

/**
 * Default props.
 *
 * @type {{useProductGraphTallyQuery: Function, children: React.ReactNode, useMetricsSelector: Function}}
 */
GraphCardMetricTotals.defaultProps = {
  children: null,
  useGraphCardContext,
  useMetricsSelector,
  useProductGraphTallyQuery
};

export { GraphCardMetricTotals as default, GraphCardMetricTotals };
