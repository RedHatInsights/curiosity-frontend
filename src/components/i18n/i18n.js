import React from 'react';
import PropTypes from 'prop-types';
import i18n from 'i18next';
import XHR from 'i18next-xhr-backend';
import { initReactI18next } from 'react-i18next';
import { helpers } from '../../common/helpers';

/**
 * ToDo: reevaluate the "I18nextProvider" on package update.
 * Appears to have timing issues, and subsequent firings of the
 * ajax/xhr setup. Reverting it to just a function call that populates behind
 * the scenes appears more predictable.
 */
class I18n extends React.Component {
  componentDidMount() {
    this.i18nInit();
  }

  componentDidUpdate(prevProps) {
    const { locale } = this.props;

    if (locale !== prevProps.locale) {
      this.i18nInit();
    }
  }

  i18nInit() {
    const { fallbackLng, loadPath, locale } = this.props;

    i18n
      .use(XHR)
      .use(initReactI18next)
      .init({
        backend: {
          loadPath
        },
        fallbackLng,
        lng: locale,
        debug: !helpers.PROD_MODE,
        ns: ['default'],
        defaultNS: 'default',
        react: {
          useSuspense: false
        }
      });
  }

  render() {
    const { children } = this.props;

    return <React.Fragment>{children}</React.Fragment>;
  }
}

I18n.propTypes = {
  children: PropTypes.node.isRequired,
  fallbackLng: PropTypes.string,
  loadPath: PropTypes.string,
  locale: PropTypes.string
};

I18n.defaultProps = {
  fallbackLng: process.env.REACT_APP_CONFIG_SERVICE_LOCALES_DEFAULT_LNG,
  loadPath: process.env.REACT_APP_CONFIG_SERVICE_LOCALES_PATH,
  locale: null
};

export { I18n as default, I18n };
