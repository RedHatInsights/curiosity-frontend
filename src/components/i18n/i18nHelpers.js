import React from 'react';
import i18next from 'i18next';
import { Trans } from 'react-i18next';
import { helpers } from '../../common/helpers';

/**
 * @memberof i18n
 * @module i18nHelpers
 */

/**
 * Check to help provide an empty context.
 *
 * @type {string}
 */
const EMPTY_CONTEXT = 'LOCALE_EMPTY_CONTEXT';

/**
 * A placeholder for "t", translation method.
 * Associated with the i18n package, and typically used as a default prop.
 *
 * @param {string|Array} key
 * @param {string|object|Array} value
 * @param {Array} components
 * @returns {string}
 */
const noopTranslate = (key, value, components) => {
  const updatedKey = (Array.isArray(key) && `[${key}]`) || key;
  const updatedValue =
    (typeof value === 'string' && value) ||
    (Array.isArray(value) && `[${value}]`) ||
    (Object.keys(value || '').length && JSON.stringify(value)) ||
    '';
  const updatedComponents = (components && `${components}`) || '';

  return `t(${updatedKey}${(updatedValue && `, ${updatedValue}`) || ''}${
    (updatedComponents && `, ${updatedComponents}`) || ''
  })`;
};

/**
 * Split a string on underscore.
 *
 * @param {string} value
 * @param {object} settings
 * @param {string} settings.emptyContextValue
 * @returns {string|string[]}
 */
const splitContext = (value, { emptyContextValue = EMPTY_CONTEXT } = {}) => {
  if (typeof value === 'string' && value !== emptyContextValue && /_/.test(value)) {
    return value.split('_');
  }
  return value;
};

/**
 * Parse extend context arrays/lists, and apply values to a concatenated translate key.
 *
 * @param {string|Array} translateKey
 * @param {*|string|Array} context
 * @param {object} settings
 * @param {string} settings.emptyContextValue
 * @param {Function} settings.splitContext
 * @returns {{translateKey: string, context: (string|string[])}}
 */
const parseContext = (
  translateKey,
  context,
  { emptyContextValue = EMPTY_CONTEXT, splitContext: aliasSplitContext = splitContext } = {}
) => {
  let updatedTranslateKey = translateKey;
  let updatedContext = context;

  if (updatedContext) {
    updatedContext = aliasSplitContext(updatedContext);
  }

  if (Array.isArray(updatedContext)) {
    const tempContext = updatedContext
      .map(value => (value === emptyContextValue && ' ') || aliasSplitContext(value))
      .flat()
      .filter(value => typeof value === 'string' && value.length > 0);

    if (tempContext?.length > 1) {
      const lastContext = tempContext.pop();

      if (Array.isArray(updatedTranslateKey)) {
        updatedTranslateKey = updatedTranslateKey.map(value => `${value}_${tempContext.join('_')}`);
      } else {
        updatedTranslateKey = `${updatedTranslateKey}_${tempContext.join('_')}`;
      }

      updatedContext = lastContext;
    } else {
      updatedContext = tempContext.join('_');
    }
  } else if (updatedContext === emptyContextValue) {
    updatedContext = ' ';
  }

  return {
    context: updatedContext,
    translateKey: updatedTranslateKey
  };
};

/**
 * Parse a translation key. If an array, filter for defined strings.
 *
 * @param {*|string|Array} translateKey
 * @returns {*}
 */
const parseTranslateKey = translateKey => {
  let updatedTranslateKey = translateKey;

  if (Array.isArray(updatedTranslateKey)) {
    updatedTranslateKey = updatedTranslateKey.filter(value => typeof value === 'string' && value.length > 0);
  }

  return updatedTranslateKey;
};

/**
 * Return a test element wrapper;
 *
 * @param {object} params
 * @param {string|Array} params.defaultTestId
 * @param {string} params.testId
 * @returns {null|React.ReactNode}
 */
const setI18nTestElement = ({ defaultTestId, testId }) => {
  if (typeof testId === 'boolean' && defaultTestId) {
    const updatedDataTest = (Array.isArray(defaultTestId) && defaultTestId[0]) || defaultTestId;
    // eslint-disable-next-line
    return function TestElementBool({ children }) {
      return (
        <span key={updatedDataTest} className="curiosity-translate__test-id" data-test={updatedDataTest}>
          {children}
        </span>
      );
    };
  }

  if (typeof testId === 'string' && testId.length > 0) {
    // eslint-disable-next-line
    return function TestElementString({ children }) {
      return (
        <span key={testId} className="curiosity-translate__test-id" data-test={testId}>
          {children}
        </span>
      );
    };
  }

  if (React.isValidElement(testId)) {
    // eslint-disable-next-line
    return function TestElementNode({ children }) {
      return React.cloneElement(testId, {}, children);
    };
  }

  return null;
};

/**
 * Apply a string towards a key. Optional replacement values and component/nodes.
 * See, https://react.i18next.com/
 *
 * @param {string|Array} translateKey A key reference, or an array of a primary key with fallback keys.
 * @param {string|object|Array} values
 *     - A default string if the key can't be found.
 *     - An object with i18next settings. i.e. "{ context: Array|string, testId: boolean|string|React.ReactNode }"
 *     - An array of objects (key/value) pairs used to replace string tokens. i.e. "[{ hello: 'world' }]"
 * @param {Array} components An array of HTML/React nodes used to replace string tokens. i.e. "[<span />, <React.Fragment />]"
 * @param {object} settings
 * @param {*} settings.i18next
 * @param {Function} settings.isDebug
 * @param {Function} settings.noopTranslate
 * @param {Function} settings.parseContext
 * @param {Function} settings.parseTranslateKey
 * @returns {string|React.ReactNode}
 */
const translate = (
  translateKey,
  values = null,
  components,
  {
    i18next: aliasI18next = i18next,
    isDebug = helpers.TEST_MODE,
    noopTranslate: aliasNoopTranslate = noopTranslate,
    parseContext: aliasParseContext = parseContext,
    parseTranslateKey: aliasParseTranslateKey = parseTranslateKey
  } = {}
) => {
  const updatedValues = values || {};
  const baseUpdatedTranslateKey = aliasParseTranslateKey(translateKey);
  let updatedTranslateKey = baseUpdatedTranslateKey;
  let TestElement;

  if (updatedValues?.context) {
    const { context: parsedContext, translateKey: parsedAgainTranslateKey } = aliasParseContext(
      updatedTranslateKey,
      updatedValues.context
    );
    updatedTranslateKey = parsedAgainTranslateKey;
    updatedValues.context = parsedContext;
  }

  if (isDebug) {
    return aliasNoopTranslate(updatedTranslateKey, updatedValues, components);
  }

  if (updatedValues?.testId) {
    TestElement = setI18nTestElement({ defaultTestId: baseUpdatedTranslateKey, testId: updatedValues.testId });
  }

  if (components && aliasI18next.store) {
    if (TestElement) {
      return (
        <TestElement>
          <Trans i18nKey={updatedTranslateKey} values={updatedValues} components={components} />
        </TestElement>
      );
    }

    return <Trans i18nKey={updatedTranslateKey} values={updatedValues} components={components} />;
  }

  if (aliasI18next.store) {
    if (TestElement) {
      return <TestElement>{aliasI18next.t(updatedTranslateKey, updatedValues)}</TestElement>;
    }

    return aliasI18next.t(updatedTranslateKey, updatedValues);
  }

  return `t([${updatedTranslateKey}])`;
};

/**
 * Apply string replacements against a component, HOC.
 *
 * @param {React.ReactNode} Component
 * @param {object} settings
 * @param {*} settings.i18next
 * @param {Function} settings.noopTranslate
 * @returns {React.ReactNode}
 */
const translateComponent = (
  Component,
  { i18next: aliasI18next = i18next, noopTranslate: aliasNoopTranslate = noopTranslate } = {}
) => {
  const withTranslation = ({ ...props }) => (
    <Component
      {...props}
      t={(aliasI18next.store && translate) || aliasNoopTranslate}
      i18n={(aliasI18next.store && aliasI18next) || helpers.noop}
    />
  );

  withTranslation.displayName = 'withTranslation';
  return withTranslation;
};

const i18nHelpers = {
  EMPTY_CONTEXT,
  noopTranslate,
  parseContext,
  parseTranslateKey,
  splitContext,
  translate,
  translateComponent
};

export {
  i18nHelpers as default,
  i18nHelpers,
  EMPTY_CONTEXT,
  noopTranslate,
  parseContext,
  parseTranslateKey,
  splitContext,
  translate,
  translateComponent
};
