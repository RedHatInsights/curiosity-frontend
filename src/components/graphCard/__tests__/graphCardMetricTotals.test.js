import React from 'react';
import { GraphCardMetricTotals } from '../graphCardMetricTotals';
import { translate } from '../../i18n/i18nHelpers';

describe('GraphCardMetricTotals Component', () => {
  it('should render a basic component', () => {
    const props = {
      children: 'lorem ipsum'
    };
    const component = renderComponent(<GraphCardMetricTotals {...props} />);

    expect(component).toMatchSnapshot('basic');
  });

  it('should handle multiple display states', async () => {
    const props = {
      useGraphCardContext: () => ({
        children: 'lorem ipsum',
        settings: {
          isMetricDisplay: true,
          groupMetric: ['hello', 'world'],
          cards: [
            {
              header: 'lorem',
              body: 'ipsum',
              footer: 'dolor sit'
            }
          ]
        }
      }),
      useMetricsSelector: () => ({
        pending: true,
        error: false,
        fulfilled: false
      })
    };
    const component = await shallowComponent(<GraphCardMetricTotals {...props} />);

    expect(component).toMatchSnapshot('pending');

    const componentPending = await component.setProps({
      useMetricsSelector: () => ({
        pending: false,
        error: true,
        fulfilled: false
      })
    });

    expect(componentPending).toMatchSnapshot('error');

    const componentFulfilled = await component.setProps({
      useMetricsSelector: () => ({
        pending: false,
        error: false,
        fulfilled: true
      })
    });

    expect(componentFulfilled).toMatchSnapshot('fulfilled');
  });

  it('should handle custom card displays', () => {
    const props = {
      useGraphCardContext: () => ({
        children: 'lorem ipsum',
        settings: {
          isMetricDisplay: true,
          groupMetric: ['hello', 'world'],
          cards: [
            {
              header: 'lorem',
              body: passedValues =>
                translate('ipsum-{{data}}', { testId: 'custom test id', data: JSON.stringify(passedValues) }),
              footer: 'dolor sit timestamp'
            },
            {
              key: 'my custom react key',
              header: passedValues => `hello-${JSON.stringify(passedValues)}`,
              body: 'world',
              footer: 'dolor sit timestamp'
            },
            {
              header: 'dolor',
              body: 'sit',
              footer: passedValues => `dolor sit timestamp-${JSON.stringify(passedValues)}`
            }
          ]
        }
      }),
      useMetricsSelector: () => ({
        pending: false,
        error: false,
        fulfilled: true
      })
    };
    const component = renderComponent(<GraphCardMetricTotals {...props} />);
    expect(component).toMatchSnapshot('fulfilled');
  });
});
