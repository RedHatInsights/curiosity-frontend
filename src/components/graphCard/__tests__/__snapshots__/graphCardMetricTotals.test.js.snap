// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphCardMetricTotals Component should handle custom card displays: fulfilled 1`] = `
<div
  className="curiosity-usage-graph__totals"
  data-test="graphMetricTotals-hello-world"
  data-test-data="{"groupMetricId":["hello","world"]}"
>
  <div>
    <div
      className="curiosity-usage-graph__totals-column"
    >
      <Card
        className="curiosity-usage-graph__totals-column-card "
        data-test="graphMetricTotalsCard-0"
        isPlain={true}
        key="3409dd31007d6ac3d52cd7bf810296df8201c75d"
      >
        <CardHeader>
          <CardTitle>
            <Title
              headingLevel="h2"
              size="md"
            >
              lorem
            </Title>
          </CardTitle>
        </CardHeader>
        <CardBody>
          <div>
            t(ipsum-{{data}}, {"testId":"custom test id","data":"{\\"groupMetricId\\":[\\"hello\\",\\"world\\"]}"})
          </div>
        </CardBody>
        <CardFooter>
          <div>
            dolor sit timestamp
          </div>
        </CardFooter>
      </Card>
      <Card
        className="curiosity-usage-graph__totals-column-card "
        data-test="graphMetricTotalsCard-1"
        isPlain={true}
        key="my custom react key"
      >
        <CardHeader>
          <CardTitle>
            <Title
              headingLevel="h2"
              size="md"
            >
              hello-{"groupMetricId":["hello","world"]}
            </Title>
          </CardTitle>
        </CardHeader>
        <CardBody>
          <div>
            world
          </div>
        </CardBody>
        <CardFooter>
          <div>
            dolor sit timestamp
          </div>
        </CardFooter>
      </Card>
      <Card
        className="curiosity-usage-graph__totals-column-card "
        data-test="graphMetricTotalsCard-2"
        isPlain={true}
        key="b21615e1e1be06472b2fd681bd40525fbfbe223d"
      >
        <CardHeader>
          <CardTitle>
            <Title
              headingLevel="h2"
              size="md"
            >
              dolor
            </Title>
          </CardTitle>
        </CardHeader>
        <CardBody>
          <div>
            sit
          </div>
        </CardBody>
        <CardFooter>
          <div>
            dolor sit timestamp-{"groupMetricId":["hello","world"]}
          </div>
        </CardFooter>
      </Card>
    </div>
  </div>
  <div>
    <div
      className="curiosity-usage-graph__totals-graph-column"
    />
  </div>
</div>
`;

exports[`GraphCardMetricTotals Component should handle multiple display states: error 1`] = `
<div
  className="curiosity-usage-graph__totals"
  data-test="graphMetricTotals-hello-world"
  data-test-data="{"groupMetricId":["hello","world"]}"
>
  <div>
    <div
      className="curiosity-usage-graph__totals-column"
    >
      <Card
        className="curiosity-usage-graph__totals-column-card blur"
        data-test="graphMetricTotalsCard-0"
        isPlain={true}
        key="3409dd31007d6ac3d52cd7bf810296df8201c75d"
      >
        <CardHeader>
          <CardTitle>
            <Title
              headingLevel="h2"
              size="md"
            />
          </CardTitle>
        </CardHeader>
        <CardBody>
          <div />
        </CardBody>
        <CardFooter>
          <div />
        </CardFooter>
      </Card>
    </div>
  </div>
  <div>
    <div
      className="curiosity-usage-graph__totals-graph-column"
    />
  </div>
</div>
`;

exports[`GraphCardMetricTotals Component should handle multiple display states: fulfilled 1`] = `
<div
  className="curiosity-usage-graph__totals"
  data-test="graphMetricTotals-hello-world"
  data-test-data="{"groupMetricId":["hello","world"]}"
>
  <div>
    <div
      className="curiosity-usage-graph__totals-column"
    >
      <Card
        className="curiosity-usage-graph__totals-column-card "
        data-test="graphMetricTotalsCard-0"
        isPlain={true}
        key="3409dd31007d6ac3d52cd7bf810296df8201c75d"
      >
        <CardHeader>
          <CardTitle>
            <Title
              headingLevel="h2"
              size="md"
            >
              lorem
            </Title>
          </CardTitle>
        </CardHeader>
        <CardBody>
          <div>
            ipsum
          </div>
        </CardBody>
        <CardFooter>
          <div>
            dolor sit
          </div>
        </CardFooter>
      </Card>
    </div>
  </div>
  <div>
    <div
      className="curiosity-usage-graph__totals-graph-column"
    />
  </div>
</div>
`;

exports[`GraphCardMetricTotals Component should handle multiple display states: pending 1`] = `
<div
  className="curiosity-usage-graph__totals"
  data-test="graphMetricTotals-hello-world"
  data-test-data="{"groupMetricId":["hello","world"]}"
>
  <div>
    <div
      className="curiosity-usage-graph__totals-column"
    >
      <Card
        className="curiosity-usage-graph__totals-column-card "
        data-test="graphMetricTotalsCard-0"
        isPlain={true}
        key="3409dd31007d6ac3d52cd7bf810296df8201c75d"
      >
        <CardHeader>
          <CardTitle>
            <Title
              headingLevel="h2"
              size="md"
            >
              <Loader
                skeletonProps={
                  {
                    "size": "lg",
                  }
                }
                tableProps={{}}
                variant="skeleton"
              />
            </Title>
          </CardTitle>
        </CardHeader>
        <CardBody>
          <div>
            <Loader
              skeletonProps={
                {
                  "height": "60px",
                  "size": "lg",
                }
              }
              tableProps={{}}
              variant="skeleton"
            />
          </div>
        </CardBody>
        <CardFooter>
          <div>
            <Loader
              skeletonProps={
                {
                  "size": "lg",
                }
              }
              tableProps={{}}
              variant="skeleton"
            />
          </div>
        </CardFooter>
      </Card>
    </div>
  </div>
  <div>
    <div
      className="curiosity-usage-graph__totals-graph-column"
    />
  </div>
</div>
`;

exports[`GraphCardMetricTotals Component should render a basic component: basic 1`] = `
<div
  className="curiosity-usage-graph__standalone"
  data-test="graphStandalone"
>
  lorem ipsum
</div>
`;
